components/dialog/dialog-confirm.js

```diff
@@ -66,12 +73,19 @@ class DialogConfirm extends DialogMixin(LitElement) {
 		`];
 	}
 
-	render() {
-		if (!this._titleId) this._titleId = getUniqueId();
-		if (!this._textId) this._textId = getUniqueId();
+	constructor() {
+		super();
+		this.critical = false;
+		this._offscreenId = getUniqueId();
+		this._textId = getUniqueId();
+		this._titleId = getUniqueId();
+	}
 
+	render() {
 		const inner = html`
+			${this.critical ? html`<d2l-offscreen id="${this._offscreenId}">${this.localize('components.dialog.critical')}</d2l-offscreen>` : nothing}
```

Comment:
The new code for `critical` is very duplicated between `d2l-dialog` and `d2l-dialog-confirm`. It all could be in the mixin but I left it out since we don't want this in the fullscreen dialog. Should I put it there anyways and somehow block usage of `critical` in fullscreen?

Reply:
Let's see where everything ends up, but we could also just create a little `renderCritical` helper in one of the files that can be exported and imported into the places that need it.

Reply:
Btw, it's also valid to reference `visibility: hidden`/`display: none` elements when using `aria-labelledby` so this is valid and avoids the extra custom element. 
```html
<div id="${this._criticalLabelId}" hidden>${this.localize('components.dialog.critical')}</div>
```